src/main/java/net/starschema/clouddb/cmdlineverification/Oauth2Bigquery.java

```diff
@@ -240,8 +233,41 @@ public static Bigquery authorizeviaservice(String serviceaccountemail,
             PrivateKey pk = getPrivateKeyFromCredentials(keypath, password);
             builder = builder.setServiceAccountPrivateKey(pk);
         }
+        credential = builder.build();
+        return credential;
+    }
 
-        GoogleCredential credential = builder.build();
+    /**
+     * This function gives back an built GoogleCredential Object from a json keyfile
+     *
+     * @param keypath
+     * @return Built GoogleCredential via and keypath
+     * @throws GeneralSecurityException
+     * @throws IOException
+     */
+    public static GoogleCredential Create_Json_Credential(String keypath) throws GeneralSecurityException, IOException {
+        logger.debug("Authorizing with service account.");
+        GoogleCredential credential = null;
+        // For .json load the key via credential.fromStream
+        File jsonKey = new File(keypath);
+        InputStream inputStream = new FileInputStream(jsonKey);
+        credential = credential.fromStream(inputStream, CmdlineUtils.getHttpTransport(), CmdlineUtils.getJsonFactory()).createScoped(GenerateScopes());
+        return credential;
+    }
+
+    /**
+     * This function gives back an Authorized Bigquery Client It uses a service
+     * account, which doesn't need user interaction for connect
+     *
+     * @param credential
+     * @return Authorized Bigquery Client via serviceaccount credential
+     * @throws GeneralSecurityException
+     * @throws IOException
+     */
+    public static Bigquery authorizeviaservice(GoogleCredential credential,
```

Comment:
We might want to not break this contract here. Perhaps authorizeviaservice should take the file (as before) and based on the file type decide which of the two methods to call (instead of making the caller decide and change the interface here)